mmengine/hooks/checkpoint_hook.py

```diff
@@ -304,6 +327,50 @@ def after_val_epoch(self, runner, metrics):
 
         self._save_best_checkpoint(runner, metrics)
 
+    def after_train(self, runner) -> None:
+        """Publish the checkpoint after train epoch.
+
+        Args:
+            runner (Runner): The runner of the training process.
+        """
+        if self.published_keys is None:
+            return
+
+        if self.save_last:
+            last_ckpt = runner.message_hub.get_info('last_ckpt')
+            assert last_ckpt ('Did not find last_checkpoint to be resumed.')
+            self._publish_model(runner, last_ckpt)
+
+        if self.save_best is not None:
+            if not self.best_ckpt_path:
+                raise RuntimeError(xxx)
+            self._publish_model(runner, best_ckpt)
```

Comment:
There are two cases. The best checkpoints may have more than 1.

https://github.com/open-mmlab/mmengine/blob/12e3d4e1e122dcd3dd183460ab2dd733f16ef26a/mmengine/hooks/checkpoint_hook.py#L415-L423